tests/test_max_weight_matching.py

```diff
@@ -0,0 +1,423 @@
+                                      42 + i, (u, v), (v, u), rx_matches,
+                                      nx_matches, list(rx_graph.edge_list())))
+
+    def test_gnp_random_against_networkx_max_cardinality(self):
+        rx_graph = retworkx.undirected_gnp_random_graph(10, .78, seed=428)
+        nx_graph = networkx.Graph(list(rx_graph.edge_list()))
+        nx_matches = networkx.max_weight_matching(
+            nx_graph, maxcardinality=True)
+        rx_matches = retworkx.max_weight_matching(
+            rx_graph, max_cardinality=True)
+        # Convert retworkx dict output to networkx set of tuples
+        for (u, v) in rx_matches:
+            if (u, v) not in nx_matches:
+                if (v, u) not in nx_matches:
+                    self.fail("seed %s failed. Element %s and it's "
+                              " reverse %s not found "
+                              "in networkx output.\nretworkx output: %s"
+                              "\nnetworkx output: %s\nedge list: %s" % (
+                                  428, (u, v), (v, u), rx_matches,
+                                  nx_matches, list(rx_graph.edge_list())))
+
+    def test_gnm_random_against_networkx(self):
+        rx_graph = retworkx.undirected_gnm_random_graph(10, 13, seed=42)
+        nx_graph = networkx.Graph(list(rx_graph.edge_list()))
+        nx_matches = networkx.max_weight_matching(nx_graph)
+        rx_matches = retworkx.max_weight_matching(rx_graph)
+        for (u, v) in rx_matches:
+            if (u, v) not in nx_matches:
+                if (v, u) not in nx_matches:
+                    self.fail("seed %s failed. Element %s and it's "
+                              " reverse %s not found "
+                              "in networkx output.\nretworkx output: %s"
+                              "\nnetworkx output: %s\nedge list: %s" % (
+                                  42, (u, v), (v, u), rx_matches,
+                                  nx_matches, list(rx_graph.edge_list())))
+
+    def test_gnm_random_against_networkx_max_cardinality(self):
+        rx_graph = retworkx.undirected_gnm_random_graph(10, 12, seed=42)
+        nx_graph = networkx.Graph(list(rx_graph.edge_list()))
+        nx_matches = networkx.max_weight_matching(
+            nx_graph, maxcardinality=True)
+        rx_matches = retworkx.max_weight_matching(
+            rx_graph, max_cardinality=True)
+        for (u, v) in rx_matches:
+            if (u, v) not in nx_matches:
+                if (v, u) not in nx_matches:
+                    self.fail("seed %s failed. Element %s and it's "
+                              " reverse %s not found "
+                              "in networkx output.\nretworkx output: %s"
+                              "\nnetworkx output: %s\nedge list: %s" % (
+                                  42, (u, v), (v, u), rx_matches,
+                                  nx_matches, list(rx_graph.edge_list())))
```

Comment:
It might be good to make a method of this part of code because it is used in other unit tests.

Reply:
Yep, that's a good call. I did this as part of: https://github.com/Qiskit/retworkx/pull/229/commits/cfd9121f0de4a282943adba878f3719c0376c5ba